pymongo/message.py

```diff
@@ -1111,6 +1112,12 @@ def _check_doc_size_limits(
         # key and the index of its namespace within ns_info as its value.
         op_doc[op_type] = ns_info[namespace]  # type: ignore[index]
 
+        # Since the data document itself is nested within the insert document
+        # it won't be automatically re-ordered by the BSON conversion.
+        # We use ChainMap here to make the _id field the first field instead.
+        if real_op_type == "insert":
+            op_doc["document"] = ChainMap(op_doc["document"], {"_id": op_doc["document"]["_id"]})  # type: ignore[index]
```

Comment:
The only reason we need to unwrap ChainMap later is because we're mutating op_doc. Instead we can do this:
```python
        doc_to_encode = op_doc
        if real_op_type == "insert":
            doc = op_doc["document"]
            if not isinstance(doc, RawBSONDocument):
                doc_to_encode = op_doc.copy()  # Shallow copy
                doc_to_encode["document"] = ChainMap(doc, {"_id": doc["_id"]})  # type: ignore[index]
        op_doc_encoded = _dict_to_bson(doc_to_encode, False, opts)
```

We also still need the RawBSONDocument here. We should add a test to ensure RawBSONDocument is not inflated after a call to bulk_write.

Reply:
Ah, I see! That's a slick solution.